### PR TITLE
Implement the recanting witness criterion 

### LINKED ISSUE #52: Implement the recanting witness criterion
See https://ftp.cs.ucla.edu/pub/stat_ser/r321-L.pdf

### PR DESCRIPTION
**Pull request by mschauer**
Fixes #52

See #52 for context.

Todo: Double-check algorithm carefully, because I never identify the witness.

### CODE REVIEW

File: src/recantingwitness.jl [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    next_open = Vector{T}()
+    next_blocked = Vector{T}()
+    
+    push!(next_open, u)
+    on_open[u] = true
+    while !isempty(next_open)
+        src = popfirst!(next_open) # get new element from queue
+        for vertex in outneighbors(g, src)
+            if !on_open[vertex]
```
Comment (archanarw) [inline src/recantingwitness.jl:19]: Removing this condition works for all the test cases. I think on_blocked for vertex `v` isn't becoming true when an open path is already found.

File: src/recantingwitness.jl [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            if !on_blocked[vertex]
+                on_blocked[vertex] = true
+                if vertex != v 
+                    push!(next_blocked, vertex)  # push onto blocked queue
+                end
+            end
+        end
+    end
+    @show on_blocked
+    @show on_open
```
Comment (mschauer) [inline src/recantingwitness.jl:46]:
```suggestion
```

File: src/recantingwitness.jl (hunk 1 above)
Comment (mschauer) [reply src/recantingwitness.jl:19]: This is about recursing twice down an open path. Think of a vertex which is reachable through two open path. Then we don't want to check twice for all path reachable. Or do I miss something?

**Comment by mschauer** [comment]
I misunderstood the role of the index `u`. I have to repeat the procedure I had before for each neighbour of `u`, not just for `u`, because `u` is not a recanting witness.

**Comment by mschauer** [comment]
I also added as tests the `kite` graph and the `act` example from the original paper

File: src/recantingwitness.jl [hunk 3]
```diff
@@ -0,0 +1,56 @@
+export has_recanting_witness
+function has_recanting_witness(g::AbstractGraph{T}, u::Integer, v::Integer,  blocked_edges::AbstractGraph{T}) where T
+    on_blocked = zeros(Bool, nv(g))
+    on_open = zeros(Bool, nv(g))
+    may_blocked = zeros(Bool, nv(g))
+    for e in edges(blocked_edges) # mark excluded vertices as seen
```
Comment (mschauer) [inline src/recantingwitness.jl:6]:
```suggestion
    for e in edges(blocked_edges) # mark sources of blocked edges
```

File: src/recantingwitness.jl [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        may_blocked[src(e)] = true
+    end
+    u == v && error("u == v")
```
Comment (mschauer) [inline src/recantingwitness.jl:9]: One could argue
```suggestion
    u == v && return false
```

**Pull request merged**